platform/admin/admin-configuration-configinstaller/src/main/java/org/codice/ddf/admin/configuration/ConfigurationInstaller.java

```diff
@@ -0,0 +1,168 @@
+    private static final Logger LOGGER = LoggerFactory.getLogger(ConfigurationInstaller.class);
+
+    private final ConfigurationAdmin configurationAdmin;
+
+    private Map<String, File> pidFileMap;
+
+    public ConfigurationInstaller(ConfigurationAdmin configurationAdmin) {
+        this.configurationAdmin = configurationAdmin;
+        this.pidFileMap = Collections.emptyMap();
+    }
+
+    /**
+     * @return a read-only map of configuration pids to felix file install properties.
+     */
+    public Map<String, File> getPidFileMap() {
+        return Collections.unmodifiableMap(pidFileMap);
+    }
+
+    public void init() throws IOException, InvalidSyntaxException {
+        this.pidFileMap = Arrays.stream(configurationAdmin.listConfigurations(null))
+                .map(FelixConfig::new)
+                .filter(config -> config.getFile() != null)
+                .collect(Collectors.toMap(FelixConfig::getPid, FelixConfig::getFile,
+                        // Use the latest value for the same key
+                        (pid1, pid2) -> pid2));
+    }
+
+    @Override
+    public void configurationEvent(ConfigurationEvent configurationEvent) {
+        try {
+            switch (configurationEvent.getType()) {
+            case CM_UPDATED:
+                LOGGER.debug("Handling update");
+                handleUpdate(new FelixConfig(() -> getConfiguration(configurationEvent)));
+                break;
+            case CM_DELETED:
+                LOGGER.debug("Handling delete");
+                handleDelete(configurationEvent.getPid());
+                break;
+            default:
+                LOGGER.debug("Unknown configuration event type, taking no action");
+            }
+        } catch (IOException e) {
+            LOGGER.debug("Error reading or modifying configurations on disk: {}", e);
+        } catch (IllegalArgumentException e) {
+            LOGGER.debug("Encountered bad arguments: {}", e);
+        }
+    }
+
+    /**
+     * Sync the config files in etc when an update occurs.
+     *
+     * @param config the {@link Configuration} wrapped in a {@link FelixConfig} that arrived as an event.
+     * @throws IOException if an issue occurs persisting to disk.
+     */
+    private void handleUpdate(FelixConfig config) throws IOException {
+        String pid = config.getPid();
+        File felixFileNew = config.getFile();
+        File felixFileOriginal = pidFileMap.get(pid);
+
+        boolean filePropChanged = config.filePropChanged(felixFileOriginal);
+
+        if (felixFileOriginal == null && felixFileNew == null) {
+            LOGGER.debug("Will not track pid {}", pid);
+            return;
+        }
+
+        if (filePropChanged) {
+            if (felixFileOriginal != null) {
+                LOGGER.debug("Felix filename prop changed, should delete file for pid {}", pid);
+                Files.delete(felixFileOriginal.toPath());
```

Comment:
I'll change it, but then we don't get the exception info in the logs for diagnosing issues.